org.adempiere.base/src/org/compiere/model/MArchive.java

```diff
@@ -342,7 +345,8 @@ public void setStorageProvider(MStorageProvider p) {
 	 * @return File - the temporary file
 	 */
 	public File saveAsZip() {
-		String name = MTable.get(Env.getCtx(), getAD_Table_ID()).getTableName() + "_" + getRecord_ID();
+		String name = MTable.get(Env.getCtx(), getAD_Table_ID()).getTableName() + "_"
+				+ (getRecord_ID() > 0 ? getRecord_ID() : getRecord_UU());

```

Comment:
Hi @hengsin - not a big deal, I tested packing out the archive for the Test_UU table, this is used for a temporary folder with very short live, the temp folder had the name:
/tmp/TestUU_8858ecc2-cf1d-405f-987f-793536037e76_9617073054397170185

the folder is just 64 bytes, even with a 27 chars tablename it would be 85 chars, very little for the supported names in linux or windows.